### ISSUE TITLE
The visible bell is very annoying, it triggers all the time and should be disabled 

### ISSUE THREAD

**Issue by mrbig033** [technomancy/better-defaults#27]
I took me a while to realize `better-defaults` was the one causing it. Quite a waste of time. Does anyone like this behavior?

**Comment by gonewest818** [comment]
I use a function that flashes only the modeline:
```
(setq visible-bell nil)

(defun my-terminal-visible-bell ()
  "A friendlier visual bell effect."
  (invert-face 'mode-line)
  (run-with-timer 0.1 nil 'invert-face 'mode-line))

(setq ring-bell-function #'my-terminal-visible-bell)
``` 

Which is not my invention, but I frankly can't remember where I first saw it.

**Comment by technomancy** [comment]
@gonewest818 that's strange; the behavior you describe is how it already works for me. Maybe this is a platform-specific problem?

**Comment by gonewest818** [comment]
hmm, FWIW I'm on macOS with Emacs installed via homebrew, which installs from https://emacsformacosx.com. The default `visible-bell` pops up a giant yellow triangle with an exclamation point over the center of the frame. It's so bad it's funny.

Oh, and the mode line flashing code I'm using is actually from [EmacsWiki: AlarmBell](https://www.emacswiki.org/emacs/AlarmBell).  The same wiki page mentions there's a `mode-line-bell` package on MELPA. Perhaps you have that package installed under the hood somewhere?

**Comment by gonewest818** [comment]
Yes, it certainly appears to be the default behavior for the Mac (NextStep) builds from the GNU source.  I found [commit 345a9c8c](http://git.savannah.gnu.org/cgit/emacs.git/commit/src?id=345a9c8cd07111a8c8367e59404688544e8b8b1d) on git.savannah.gnu.org showing the original implementation, and what's on master looks fundamentally the same.

There is the so-called "railwaycat" fork of Emacs for Cocoa which I don't use, but it seems to have a different implementation. Appears to be the "inverting a rectangle in the middle of the frame" technique that is mentioned in the description of the git commit above.

**Comment by mrbig033** [comment]
I fixed with this:

``` emacs-lisp
(use-package better-defaults
  :config
  (setq visible-bell nil))
```

**Comment by gonewest818** [comment]
Yes, definitely, that will “un-set” the setting that better-defaults put in place.
